generator-mediator-js/app/templates/mediatorTemplate/lib/_index.js

```diff
@@ -0,0 +1,116 @@
+
+// socket config - large documents can cause machine to max files open
+const https = require('https')
+const http = require('http')
+
+https.globalAgent.maxSockets = 5
+http.globalAgent.maxSockets = 5
+
+var port = mediatorConfig.endpoints[0].port
+
+/**
+ * setupApp - configures the http server for this mediator
+ *
+ * @return {express.App}  the configured http server
+ */
+function setupApp () {
+  const app = express()
+
+  app.get(mediatorConfig.endpoints[0].path, (req, res) => {
+    console.log('Processing GET request on %s', mediatorConfig.endpoints[0].path)
+    var responseBody = 'Primary Route Reached'
+    var headers = { 'content-type': 'application/json' }
+
+    // capture orchestration data
+    var orchestrationResponse = { statusCode: 200, headers: headers }
+    let orchestrations = []
+    orchestrations.push(utils.buildOrchestration('Primary Route', new Date().getTime(), req.method, req.url, req.body, orchestrationResponse, responseBody))
+
+    // set content type header so that OpenHIM knows how to handle the response
+    res.set('Content-Type', 'application/json+openhim')
+
+    // construct return object
+    var properties = { property: 'Primary Route' }
+    res.send(utils.buildReturnObject(mediatorConfig.urn, 'Successful', 200, headers, responseBody, orchestrations, properties))
+  })
+  return app
+}
+
+/**
+ * start - starts the mediator
+ *
+ * @param  {Function} callback a node style callback that is called once the
+ * server is started
+ */
+function start (callback) {
+  if (apiConf.api.trustSelfSigned) { process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0' }
+
+  if (apiConf.register) {
+    medUtils.registerMediator(apiConf.api, mediatorConfig, (err) => {
+      if (err) {
+        console.log('Failed to register this mediator, check your config')
+        console.log(err.stack)
+        process.exit(1)
+      }
+      apiConf.api.urn = mediatorConfig.urn
+      medUtils.fetchConfig(apiConf.api, (err, newConfig) => {
+        console.log('Received initial config:')
+        console.log(JSON.stringify(newConfig))
+        config = newConfig
+        if (err) {
+          console.log('Failed to fetch initial config')
+          console.log(err.stack)
+          process.exit(1)
+        } else {
+          console.log('Successfully registered mediator!')
+          let app = setupApp()
+          const server = app.listen(port, () => {
+            let configEmitter = medUtils.activateHeartbeat(apiConf.api)
+            configEmitter.on('config', (newConfig) => {
+              console.log('Received updated config:')
+              console.log(JSON.stringify(newConfig))
+              // set new config for mediator
+              config = newConfig
+              // example of the fun things we can do:
+              // edit default channel with new config
+              const openhim = OpenHIM(apiConf.api)
+              openhim.fetchChannelByName(mediatorConfig.defaultChannelConfig[0].name, (err, channel) => {
```

Comment:
I think this can be remove, not all mediators will need this. We can replace it with a comment that explains the type of things a developer could do here.